### PR TITLE
Fix set cache_root from config file error #8515

### LINKED ISSUE #8515: Setting cache_root from config or env gives error
Before you start, are you using the latest CocoaPods release? Yes, 1.5.3
 [X ] I've read and understood the [*CONTRIBUTING* guidelines and have done my best effort to follow](https://github.com/CocoaPods/CocoaPods/blob/master/CONTRIBUTING.md).

# Report

## What did you do?
Edit ~/.cocapods/config.yaml and added:
cache_root: "/Users/username/testcache"
Run: `pod cache list`

## What did you expect to happen?
Empty cache info displayed

## What happened instead?
Error displayed

```
NoMethodError - undefined method `exist?' for ".":String
/Users/piershearn/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/cocoapods-1.5.3/lib/cocoapods/config.rb:90:in `cache_root'
/Users/piershearn/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/cocoapods-1.5.3/lib/cocoapods/command/cache.rb:17:in `initialize'
/Users/piershearn/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/cocoapods-1.5.3/lib/cocoapods/command/cache/list.rb:25:in `initialize'
/Users/piershearn/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/claide-1.0.2/lib/claide/command.rb:356:in `new'
/Users/piershearn/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/claide-1.0.2/lib/claide/command.rb:356:in `parse'
/Users/piershearn/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/claide-1.0.2/lib/claide/command.rb:352:in `parse'
/Users/piershearn/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/claide-1.0.2/lib/claide/command.rb:352:in `parse'
/Users/piershearn/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/claide-1.0.2/lib/claide/command.rb:330:in `run'
/Users/piershearn/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/cocoapods-1.5.3/lib/cocoapods/command.rb:52:in `run'
/Users/piershearn/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/cocoapods-1.5.3/bin/pod:55:in `<top (required)>'
/Users/piershearn/.rbenv/versions/2.4.2/bin/pod:23:in `load'
/Users/piershearn/.rbenv/versions/2.4.2/bin/pod:23:in `<main>'
```

### Stack

```
   CocoaPods : 1.5.3
        Ruby : ruby 2.4.2p198 (2017-09-14 revision 59899) [x86_64-darwin16]
    RubyGems : 2.6.13
        Host : Mac OS X 10.13.6 (17G4015)
       Xcode : 10.1 (10B61)
         Git : git version 2.20.1
Ruby lib dir : /Users/****/.rbenv/versions/2.4.2/lib
Repositories : master - https://github.com/CocoaPods/Specs.git @ aadeaf5e6e2f1b7fba720444076268b9b9b5a4ec
```

### Installation Source

```
Executable Path: /Users/****/.rbenv/versions/2.4.2/bin/pod
```

### Plugins

```
cocoapods-deintegrate : 1.0.2
cocoapods-plugins     : 1.0.0
cocoapods-search      : 1.0.0
cocoapods-stats       : 1.0.0
cocoapods-trunk       : 1.3.1
cocoapods-try         : 1.1.0
slather               : 2.4.6
```

### Podfile
Not relevant, happens with all projects

## Project that demonstrates the issue
Not relevant, happens with all projects

### PR DESCRIPTION
**Pull request by tripleCC**
closes #8515 

cache_root read from config.yaml is a string

### CODE REVIEW

**Comment by CocoaPodsBarista** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please include a CHANGELOG entry to credit yourself! <br />
You can find it at <a href="https://github.com/CocoaPods/CocoaPods/blob/master/CHANGELOG.md">CHANGELOG.md</a>.</td>
    </tr>
  </tbody>
</table>

Here's an example of your CHANGELOG entry:

```markdown
* Fix set cache_root from config file error #8515  
  [tripleCC](https://github.com/tripleCC)
  [#issue_number](https://github.com/CocoaPods/CocoaPods/issues/issue_number)
```

*note*: There are two invisible spaces after the entry's text.


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: CHANGELOG.md [hunk 1]
```diff
@@ -76,6 +76,10 @@ To install release candidates run `[sudo] gem install cocoapods --pre`
 
 ##### Bug Fixes
 
+* Fix set cache_root from config file error
```
Comment (segiddins) [inline CHANGELOG.md]: Please put this in the `Master` section?

**Comment by tripleCC** [comment]
@segiddins  fix issue #8515

File: CHANGELOG.md (hunk 1 above)
Comment (tripleCC) [reply CHANGELOG.md]: done

File: CHANGELOG.md [hunk 2]
```diff
@@ -12,7 +12,9 @@ To install release candidates run `[sudo] gem install cocoapods --pre`
 
 ##### Bug Fixes
 
-* None.  
+* Fix set cache_root from config file error  
+  [tripleCC](https://github.com/tripleCC)
+  [#8551](https://github.com/CocoaPods/CocoaPods/pull/8551)
```
Comment (dnkoutso) [inline CHANGELOG.md]: nit: link to https://github.com/CocoaPods/CocoaPods/issues/8515 instead?

File: CHANGELOG.md [hunk 3]
```diff
@@ -12,7 +12,9 @@ To install release candidates run `[sudo] gem install cocoapods --pre`
 
 ##### Bug Fixes
 
-* None.  
+* Fix set cache_root from config file error
```
Comment (dnkoutso) [inline CHANGELOG.md]: very nit: wrap \`cache_root\` with backticks

**Comment by dnkoutso** [comment]
@tripleCC can you rebase so we can land this?

**Comment by tripleCC** [comment]
@dnkoutso sorry for late reply
done

File: CHANGELOG.md [hunk 4]
```diff
@@ -72,6 +72,10 @@ To install release candidates run `[sudo] gem install cocoapods --pre`
 * Do not link specs into user targets that are only used by app specs.  
   [Samuel Giddins](https://github.com/segiddins)
 
+* Fix set `cache_root` from config file error
```
Comment (dnkoutso) [inline CHANGELOG.md]: need to move this under master section since beta.3 is out.

**Pull request merged**